conan_provider.cmake

```diff
@@ -14,18 +14,51 @@ function(detect_os OS OS_VERSION)
         else()
             set(${OS} ${CMAKE_SYSTEM_NAME} PARENT_SCOPE)
         endif()
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+            string(REPLACE "-" ";" PLATFORM_LIST ${ANDROID_PLATFORM})
+            list(GET PLATFORM_LIST 1 _OS_API_LEVEL)
+            message(STATUS "CMake-Conan: android_platform=${ANDROID_PLATFORM}")
+            set(${OS_API_LEVEL} ${_OS_API_LEVEL} PARENT_SCOPE)
+        endif()
+    endif()
+endfunction()
+
+
+function(detect_arch OS ARCH)
+    if(${OS} STREQUAL "Android")
+        if(CMAKE_ANDROID_ARCH_ABI STREQUAL arm64-v8a)
```

Comment:
Is `CMAKE_SYSTEM_PROCESSOR` not already correctly set by now? 
The legacy Android toolchain does the following mapping:
* `arm64-v8a` -> `aarch64`
* `armeabi-v7a` -> `armv7-a`
* `x86` -> `i686`

Only saying this because we could cover more ground (i.e. cross-building on Linux, not just Android) if we use the values from `CMAKE_SYSTEM_PROCESSOR` instead.

Reply:
Yes, they are set as you described so I agree, this would be a more general way to set Conan's `arch` for more than just Android. I can do that here or in a new PR.

Reply:
I ended up doing this in this PR, works great!